card.py

```diff
@@ -2,7 +2,28 @@ class Card:
     def __init__(self, v, s):
         self.value = v
         self.suit = s
+        self.score = self.get_score()
 
     def __repr__(self):
-        return f'{self.value} of {self.suit}'
+        return f'{self.get_value_name()}{self.suit}'
 
+    def get_score(self):
+        if self.value > 10:
+            return 10
+        elif self.value == 1:
+            return 11
+        else:
+            return self.value
+
+    def get_value_name(self):
+        match self.value:
```

Comment:
This is not a bad solution. To note: I don't think I've ever used a match/case in python. I had to google the syntax to make sure this was correct. There is a much cleaner version of this using dicts.

```python
face_value_mapping = dict(
    1='A',
    11='J',
    12='Q',
    13='K',
)
...
return self.face_value_mapping.get(self.value, self.value)
```

I would add `face_value_mapping` as an attribute on the class between lines 1 and 2. Much cleaner, and then we don't need to worry about match/case. It's also a more pythonic solution to this issue.

Reply:
In C# `self` is `this`. `self`.

The `.get` method on dictionary is a helper method around grabbing items out of the dictionary. The first argument is what key to grab, the second argument is what to use if the key is not present. So, lets say the card is the 6 of diamonds. We'll call `face_value_mapping.get(self.value, self.value)`. `self.value` is 6 because that's what this instance has it defined as. `6` does not show up in the dictionary, so fall back to returning `6`. Now, if The card is the A of Spades, `self.value == 1`. In that case, `self.face_value_mapping.get(self.value, self.value)` would not the second argument. It would find the key/value mapping of `1='A'` and return the `A`. Does that make sense?

Reply:
Ahhhh, that's simple. In Visual Studio, I could view the definition of methods or objects to understand them better. I'm sure I can do that in Pycharm, but I haven't found it yet. Thank you.

Reply:
It's easier to find the definition of methods in a strongly typed language. Python is loosely typed. My money says that PyCharm can find definitions of methods that are part of the current project, but `dict.get` is part of the standard lib. In the future, I can try to link to relevant documentation when throwing something new from the standard lib is involved.